### PR TITLE
Nested folders: Unify visual styles between tree + search view

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- **IMPORTANT NOTE**: this is not feature toggled so will modify the old search view as well.
- unifies the table styling between the browse tree view and the browse search view
  - adjusts row height of tree view from 35 to 36 to match search view
  - removes lots of custom cell padding in favour of the more universal flex/cell styling in tree view
  - adds skeleton loading rows to replace spinner when first opening search view
  - adds skeleton loading rows when scrolling the search view
    - this only really affects searchV2 for now since that's the one that's paginated
  - replaces the janky select all checkbox with the indeterminate checkbox
    - this is **almost** exactly the same behaviour as in tree view.
    - **HOWEVER** there is one difference! there is no way to tell if **everything** is selected currently, so in this case it will show an indeterminate checkbox instead of fully selected.
    - we can maybe fix this in a follow up PR? either way, i think this is significantly better than the old checkbox (which had the opposite problem of showing everything selected when it was not) 😅 
    
# before (note the checkbox icon having to load 🤮 )

https://github.com/grafana/grafana/assets/20999846/766847f5-eff9-4e0c-8aa5-4a65810988b1

# after

https://github.com/grafana/grafana/assets/20999846/cccc824a-204c-4e80-8672-bbd9f9fa4e71

**Why do we need this feature?**

- better UX

**Who is this feature for?**

- everyone!

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/search/page/components/columns.tsx [hunk 1]
```diff
@@ -127,20 +119,26 @@ export const generateColumns = (
...
     },
     id: `column-name`,
     field: access.name!,
-    Header: () => <div className={styles.headerNameStyle}>{t('search.results-table.name-header', 'Name')}</div>,
+    Header: () => <div>{t('search.results-table.name-header', 'Name')}</div>,
     width,
   });
   availableWidth -= width;
 
   width = TYPE_COLUMN_WIDTH;
```
Comment (joshhunt) [inline public/app/features/search/page/components/columns.tsx:140]: I think we can simplify this width logic considerably more. 

Previously this was laying things out with pixels (using the container width that the auto-sizer gets), but now this custom flex layout uses relative values, so I don't think we need to do this whole `availableWidth -= width` stuff any more? Each column can just have a static number that'll work out (like how it works in the tree)

**Review by joshhunt — COMMENTED** [review]
looks really nice. Great to see less things jumping about when switching between browse and search.

I do have a little concern around the column sizing in SearchView?

**Review by joshhunt — APPROVED** [review]

File: public/app/features/search/page/components/columns.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/features/search/page/components/columns.tsx:140]: @joshhunt one extra thing it does do is remove columns that don't have enough width remaining. this is helpful for smaller viewports (e.g. mobile). in that case the `tags`/`location` columns are completely removed. so whilst we can simplify, think we can hold off doing that until the unification work between this table and the `DashboardsTree` table. wdyt? 🤔

**Comment by joshhunt** [comment]
Sounds good @ashharrison90 :shipit:

**Pull request merged**